### DISCUSSION TITLE
import () in "Code generated by sqlc"

### DISCUSSION THREAD

**Discussion by zoranembroker** [sqlc-dev/sqlc#1402] [Q&A]
It is not clear how "import" is generated in sqlc.

E.g.
import (
	"context"
	"encoding/json"
	"time"
)

I have a query that is using a column which is 'timestamp' postgres type.

Sqlc generate makes the proper struct with 'time.Time' type.

E.g.
StartedAt                    *time.Time

However, in the import "time" is missing :
import (
	"context"
	"encoding/json"
)

Not sure what to do to make it inserted there?

**Comment by sxwebdev** [comment]
try to use https://github.com/tkcrm/pgxgen

```bash
pgxgen sqlc generate
```

**Comment by Jille** [comment]
sqlc should determine the imports for you, but there might be a bug.

Can you show a more concrete example on https://play.sqlc.dev/?

**Comment by jhbrown94** [comment]
I am able to duplicate this behavior by providing a RETURNING clause on a query marked `:exec`, e.g.
https://play.sqlc.dev/p/243aae9aa9c5ad40c6a85f8b183f96635412965acf280be3e7e7e4f01abd8ab9

It's pretty nonsensical to have a RETURNING clause there, but it would be nice to get a warning from sqlc instead of an un-compilable file.
